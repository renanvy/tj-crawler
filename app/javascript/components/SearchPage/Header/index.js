import React, { Component } from "react";
import PropTypes from "prop-types";

export default class Header extends Component {
  constructor(props) {
    super(props);

    this.state = {
      processNumber: ""
    };
  }

  componentDidMount() {
    this.startsMaterializeSelect();
  }

  startsMaterializeSelect() {
    const materializeSelect = document.querySelectorAll("select");
    M.FormSelect.init(materializeSelect, {});
  }

  render() {
    const { loading, search } = this.props;
    const { processNumber } = this.state;

    return (
      <header>
        <form onSubmit={evt => search(evt, processNumber)}>
          <div className="row">
            <div className="input-field col s12 m3">
              <select defaultValue="tjal">
                <option value="tjal" disabled>
                  TJAL
                </option>
              </select>
              <label>Tribunal</label>
            </div>

            <div className="input-field col s12 m5">
              <input
                id="process_number"
                type="text"
                className="validate"
                value={processNumber}
                onChange={evt =>
                  this.setState({
                    processNumber: evt.target.value
                  })
                }
              />

              <label htmlFor="process_number">NÃºmero do Processo</label>
            </div>

            <button
              style={{ marginTop: 5 }}
              type="submit"
              className="btn btn-large"
              disabled={loading || !processNumber}
            >
              <i className="material-icons">search</i>
            </button>
          </div>
        </form>
      </header>
    );
  }
}

Header.propTypes = {
  loading: PropTypes.bool.isRequired,
  search: PropTypes.func.isRequired
};
