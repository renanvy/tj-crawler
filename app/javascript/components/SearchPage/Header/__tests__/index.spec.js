import React from "react";
import renderer from "react-test-renderer";
import Header from "../index";
import "materialize-css/dist/js/materialize.min";

describe("Header", () => {
  test("snapshot renders", () => {
    const component = renderer.create(<Header loading search={() => {}} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
