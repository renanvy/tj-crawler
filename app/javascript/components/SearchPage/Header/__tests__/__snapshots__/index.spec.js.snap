// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot renders 1`] = `
<header>
  <form
    onSubmit={[Function]}
  >
    <div
      className="row"
    >
      <div
        className="input-field col s12 m3"
      >
        <select
          defaultValue="tjal"
        >
          <option
            disabled={true}
            value="tjal"
          >
            TJAL
          </option>
        </select>
        <label>
          Tribunal
        </label>
      </div>
      <div
        className="input-field col s12 m5"
      >
        <input
          className="validate"
          id="process_number"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="process_number"
        >
          NÃºmero do Processo
        </label>
      </div>
      <button
        className="btn btn-large"
        disabled={true}
        style={
          Object {
            "marginTop": 5,
          }
        }
        type="submit"
      >
        <i
          className="material-icons"
        >
          search
        </i>
      </button>
    </div>
  </form>
</header>
`;
